src/components/full-screen-popover/FullScreenPopover.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import * as React from 'react';
+import uniqueId from 'lodash/uniqueId';
+
+import classNames from 'classnames';
+import { Overlay } from '../flyout';
+import { KEYS } from '../../constants';
+import OverlayHeader from './OverlayHeader';
+import './FullScreenPopover.scss';
+
+export type FullScreenPopoverProps = {
+    /** Components to render in the overlay */
+    children: React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Content nested inside of the popover header */
+    header?: React.Element<any>,
+    /** Fires this callback when overlay is closed */
+    onClose?: Function,
+    /** CSS selector that identifies clickable elements that closes the overlay */
+    onCloseClassSelector?: string,
+    /** Fires this callback when overlay is opened */
+    onOpen?: Function,
+};
+
+type Props = FullScreenPopoverProps;
+
+const FullScreenPopover = (props: Props) => {
+    const [isVisible, setVisibility] = React.useState(false);
+
+    const overlayID = uniqueId('overlay');
+    const popoverButtonID = uniqueId('popoverbutton');
+
+    const { className, header, children, onClose, onCloseClassSelector, onOpen } = props;
+
+    const elements = React.Children.toArray(children);
+
+    if (elements.length !== 2) {
+        throw new Error('FullScreenPopover must have exactly two children: A button component and a content element');
+    }
+
+    const [popoverButton, overlayContent] = elements;
+
+    const openOverlay = () => {
+        setVisibility(true);
+        if (onOpen) {
+            onOpen();
+        }
+    };
+
+    const closeOverlay = () => {
+        setVisibility(false);
+        if (onClose) {
+            onClose();
+        }
+    };
+
+    const focusButton = () => {
+        const buttonEl = document.getElementById(popoverButtonID);
+        if (buttonEl) {
+            buttonEl.focus();
+        }
+    };
+
+    const handleOverlayClick = (event: SyntheticEvent<>) => {
+        // Searches overlay for classes that match event.target and closes overlay if match
+        const overlayNode: HTMLElement | null = document.getElementById(overlayID);
```

Comment:
In theory you should be able to use a direct ref to the element instead of this query